File: designPattern/Flyweight/basic/BigChar.java
import java.io.*;

public class BigChar{
	private String fontdata;

	public BigChar(char charname){
		try{
			BufferedReader reader = new BufferedReader(
				new FileReader("big"+charname+".txt")
			);
			String line;
			StringBuilder buf = new StringBuilder();
			while((line = reader.readLine()) != null){
				buf.append(line);
				buf.append("\n");
			}
			reader.close();
			this.fontdata = buf.toString();
		}catch(IOException e){
			this.fontdata = charname+"?";
		}
	}

	public void print(){
		System.out.print(fontdata);
	}
}

File: designPattern/Flyweight/pythonBasic/BigChar.py
class BigChar:

    def __init__(self, number):
        self.txt = ""
        with open("big" + number + ".txt", 'r') as fp:
            lines = fp.read()
            for line in lines:
                self.txt += line

    def getBigChar(self):
        return self.txt

